src/backend/d3d12/CommandBufferD3D12.cpp

```diff
@@ -44,8 +187,59 @@ namespace d3d12 {
     }
 
     void CommandBuffer::FillCommands(ComPtr<ID3D12GraphicsCommandList> commandList) {
+        BindGroupStateTracker bindingTracker(device);
```

Comment:
As mentioned offline what we can do for now for FillCommands is:

 - Allocate a big descriptor heap (and assume we won't overflow in the rest of FillCommands)
 - On set bind groups, if the bindgroup isn't in this heap already, record it in place (this requires a Serial heapSerial in the bindgroup) and store the offset in the bindgroup.

Then in the actual execution of commands we can just ask the bindgroup for its offset, and update the root table.